felix/rules/endpoints.go

```diff
@@ -385,31 +413,31 @@ func (r *DefaultRuleRenderer) endpointIptablesChain(policyGroups []*PolicyGroup,
 
 	// First set up failsafes.
 	if failsafeChain != "" {
-		rules = append(rules, Rule{
-			Action: JumpAction{Target: failsafeChain},
+		rules = append(rules, generictables.Rule{
+			Match:  r.NewMatch(),
```

Comment:
One of the unfortunate consequences of using an interface type for the match is that leaving the `Match` type `nil` is no longer viable, as it results in nil pointer exceptions when calling bound functions where previously the function would have been called on a `nil` slice.

Reply:
Yeah, I was trying to think of a way around that.  Probably not worth it, but you could maybe do it with generics:
```
type Rule[M Match, A Action] {
  Match M
  Action A
}
```
Then the rule renderer could have `[M Match, A Action]` too and pass them through.  Would create a different kind of boilerplate everywhere!  Advantage would be that the compiler would enforce getting it right.